### DISCUSSION TITLE
How to use blink's capabilities for all lsp servers?

### DISCUSSION THREAD

**Discussion by OneOfOne** [LazyVim/LazyVim#4861] [Q&A] [answered]
I tried this, however I get 
```Failed to run `config` for nvim-lspconfig

/home/oneofone/.config/nvim/lua/plugins/code.lua:57: attempt to call field 'setup' (a nil value)

```

If I remove the loop, it works fine.

```lua
		opts = function(_, opts)
			local ret = vim.tbl_deep_extend('force', opts, {
				format_notify = true,
				inlay_hints = {
					enabled = false,
				},
				codelens = {
					enabled = true,
				},
				showMessage = {
					messageActionItem = {
						additionalPropertiesSupport = true,
					},
				},
				format = {
					timeout_ms = 1000,
				},
				servers = {
					jsonls = {
						commands = {
							Format = {
								function() end,
							},
						},
						json = {
							format = {
								enable = false,
							},
						},
					},
					biome = {},
					html = {},
				},
			})
			local lspconfig = require('lspconfig')
			for server, config in pairs(ret.servers or {}) do
				vim.notify('Setting up ' .. server .. ' server')
				config.capabilities = require('blink.cmp').get_lsp_capabilities(config.capabilities)
				lspconfig[server].setup(config)
			end
			return ret
		end,
```

**Comment by folke** [comment]
Just use the blink extra?

**Reply by OneOfOne** [reply]
The blink extra doesn't setup that.

**Reply by OneOfOne** [reply]
I'm using it

**Reply by folke** [reply, marked as answer]
It does. it happens in the lsp plugin, so just remove all you have here and it will work.

**Reply by denesgeza** [reply]
It doesn't work for me also with the blink extra but only for certain filetypes like ts, js, html for all other works perfectly.
I don't have any lspconfig or blink options setup.

**Reply by folke** [reply]
What specifically does not work?

**Reply by folke** [reply]
https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/lsp/init.lua#L191

**Reply by denesgeza** [reply]
Completion from blink does not appear only ghost text from copilot. Servers are emmet_language_server and vtsls.

**Reply by folke** [reply]
Then you must have something in your config that breaks it, since it all works as it should for me in those languages.
Remove all your blink/snippet/`<tab>` customizations and make sure you use the blink/copilot extra.

**Reply by denesgeza** [reply]
Thanks a lot for the time investigating. Appreciate the work!
